policysync/processor.go

```diff
@@ -372,10 +399,68 @@ func (p *Processor) handleNamespaceRemove(update *proto.NamespaceRemove) {
 	delete(p.namespaceByID, id)
 }
 
+func (p *Processor) handleIPSetUpdate(update *proto.IPSetUpdate) {
+	id := update.Id
+	logCxt := log.WithField("ID", id)
+	logCxt.Debug("Processing IPSetUpdate")
+	s, ok := p.ipSetsByID[id]
+	if !ok {
+		logCxt.Info("Adding new IP Set")
+		s = newIPSet(update)
+		p.ipSetsByID[id] = s
+
+		// Since the calc graph will always send IPSets before any policies/profiles
+		// that reference them, we don't know which endpoints will need this IPSet yet,
+		// so we can stop processing here.
+		return
+	}
+	logCxt.Info("Updating existing IPSet")
+	s.replaceMembers(update)
+
+	// gRPC has limits on message size, so break up large update if necessary.
+	updates := splitIPSetUpdate(update)
+	for _, ei := range p.updateableEndpoints() {
+		if p.referencesIPSet(ei, id) {
+			ei.syncedIPSets[id] = true
+			for _, u := range updates {
+				ei.output <- u
+			}
+		}
+	}
+}
+
+func (p *Processor) handleIPSetDeltaUpdate(update *proto.IPSetDeltaUpdate) {
+	id := update.Id
+	log.WithField("ID", id).Debug("Processing IPSetDeltaUpdate")
+
+	// We trust the calc graph to never send us Delta updates for non-existent sets.
+	s := p.ipSetsByID[id]
+	s.deltaUpdate(update)
+
+	// gRPC has limits on message size, so break up large update if necessary.
+	updates := splitIPSetDeltaUpdate(update)
+	for _, ei := range p.updateableEndpoints() {
```

Comment:
There are enough subtle differences that I don't want to do that.

In the `IPSetUpdate` version, we also add the IPSet ID to a set.

I don't want to use `sendIPSetUpdate(ei *EndpointInfo, id string)` because I want the splitting to happen just once, outside the endpoint loop.